docs/development/pigs/pig-009.rst

```diff
@@ -0,0 +1,231 @@
+
+Then, the time will be sampled independently. This will be done through a 
+`LightCurveTableModel.sample(n_events=, random_state=)` method that we propose to add 
+into `~gammapy.time.models.LightCurveTableModel` or `~gammapy.time.models.PhaseCurveTableModel`. 
+This method will take as input a lightcurve (or phasecurve) model and the GTIs (i.e. one 
+Tstart and Tstop) in the `MapDataset` object. Also in this case the `InverseCDFSamler` 
+class is the machine used to sample the time of the events. In the case the temporal 
+model is not provided, the time is uniformly sampled in the time range Tstart-Tstop.
+
+The IRF correction can now be applied to sampled events. We propose to add a 
+`EdispMap.distribute(events=)` method in both `~gammapy.cube.PSFMap` and `~gammapy.cube.EdispMap`. 
+The method interpolates the "correct" IRF at the position of a given event and
+applies it. In more detail, the method calculates the psf and the energy dispersion at
+the events true positions and true energies, which are given in input as an `Astropy.Table` object.
+The IRFs are assumed to be constant and not time-dependent. The output will be an `Astropy.Table` 
+with the new columns `RA`, `DEC` and `ENERGY`, which are the reconstructed event energies and 
+positions.
+
+Finally, the times and the energies/coordinates of the events will be merged into a 
+unique `~astropy.table.Table` with the columns:`RA`, `DEC` and `ENERGY and `TIME` .
+ 
+The `MapDatasetEventSampler` can be used to sample background events using the 
+`Map.sample(n_events=, random_state=)` as well. The time of the events is sampled
+assuming a costant event rate. Finally, the IRF corrections are not applied to sampled
+events.
+
+
+Performance and precision evaluation
+====================================
+To evaluate the precision and performance of the described framework we propose
+to implement a prototype for a simulation / fitting pipeline. Starting from a
+selection of spatial, spectral and temporal models, data is simulated and fitted
+multiple times to derive distributions and pull-distributions of the reconstructed
+model parameters. This pipeline should also monitor the required cpu and memory usage.
+This first prototype can be used to evaluate the optimal bin-size (with the best
+compromise between performance and precision) for the simulations and to verify
+the over-all correctness of the simulated data. Later this prototype can be
+developed further into a full simulation / fitting continuous integration system.
+
+
+Alternatives / Outlook
+======================
+So far Gammapy only supports binned likelihood analysis and technically most use-
+cases for the event sampling could be solved with binned simulations. A binned
+simulation can be basically achieved by a call to `numpy.random.poisson()` based
+on the predicted number of counts map. This is conceptionally simpler as well as
+computationally more efficient then a sampling of event lists. However, to support
```

Comment:
than a sampling.

You could say that this reduced dataset simulation already exist with `Dataset.fake()` but has a more limited use case than event sampling.

Reply:
Done.